core/src/main/java/org/jruby/RubyFile.java

```diff
@@ -1638,32 +1638,42 @@ private static IRubyObject expandPathInternal(ThreadContext context, IRubyObject
             int index = relativePath.indexOf("!/");
             postFix = relativePath.substring(index);
             relativePath = relativePath.substring(0, index);
-        }
-        else if (protocol.find()) {
+        } else if (protocol.find()) {
             preFix = protocol.group();
             int offset = protocol.end();
             String extra = "";
             int index = relativePath.indexOf("file://");
+            boolean classloaderURI = preFix.equals("uri:classloader:") || preFix.equals("classpath:");
+
             if (index >= 0) {
                 index += 7; // "file://".length == 7
                 // chck if its "file:///"
                 if (relativePath.length() > index && relativePath.charAt(index) == '/') {
                     offset += 2; extra = "//";
-                }
-                else {
+                } else {
                     offset += 1; extra = "/";
                 }
+            } else {
+                if (classloaderURI && relativePath.startsWith("//", offset)) {
+                    // on Windows "uri:classloader://home" ends up as "//home" - trying a network drive!
+                    offset += 1; // skip one '/'
+                }
             }
-            else if ( ( preFix.equals("uri:classloader:") || preFix.equals("classpath:") )
-                    && relativePath.startsWith("//", offset) ) {
-                // on Windows "uri:classloader://home" ends up as "//home" - trying a network drive!
-                offset += 1; // skip one '/'
+
+            relativePath = relativePath.substring(offset);
+
+            if (classloaderURI) {
+                String fakePrefix = "/THIS_IS_A_FAKE_PATH_FOR_JRUBY";
+                relativePath = canonicalizePath(fakePrefix + "/" + relativePath).substring(fakePrefix.length());
```

Comment:
@headius this ends up being off by two characters on Windows because the canonicalized path with begin with `C:`. So you end up with `classpath:BY\yourpathhere.rb". Sorry I couldn't test ahead of time.